packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -78,6 +172,45 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
         });
     }
 
+    private async performOwnSharedIdentityAttributeSuccession(sharedPredecessorId: CoreId, sourceSuccessor: LocalAttribute, requestInfo: LocalRequestInfo) {
+        const successorParams = {
+            content: sourceSuccessor.content,
+            shareInfo: LocalAttributeShareInfo.from({
+                peer: requestInfo.peer,
+                requestReference: requestInfo.id,
+                sourceAttribute: sourceSuccessor.id
+            })
+        };
+        const { successor } = await this.consumptionController.attributes.succeedOwnSharedIdentityAttribute(sharedPredecessorId, successorParams);
+        return successor;
+    }
+
+    private async performOwnSharedThirdPartyRelationshipAttributeSuccession(sharedPredecessorId: CoreId, sourceSuccessor: LocalAttribute, requestInfo: LocalRequestInfo) {
+        const successorParams = {
+            content: sourceSuccessor.content,
+            shareInfo: LocalAttributeShareInfo.from({
+                peer: requestInfo.peer,
+                requestReference: requestInfo.id,
+                sourceAttribute: sourceSuccessor.id
+            })
+        };
+        const { successor } = await this.consumptionController.attributes.succeedOwnSharedRelationshipAttribute(sharedPredecessorId, successorParams);
```

Comment:
I understand the advantage of encapsulation, but it also seems a bit redundant to me to define the three functions `performOwnSharedIdentityAttributeSuccession`, `performOwnSharedThirdPartyRelationshipAttributeSuccession` and `performThirdPartyOwnedRelationshipAttributeSuccession`, which have essentially identical code, instead of calling the corresponding AttributesController.ts functions directly above. If the proposed refactoring takes place, consider instead defining a function `succeedOwnSharedThirdPartyRelationshipAttribute` in the AttributesController.ts that works like the `succeedOwnSharedRelationshipAttribute` function (so that it is still emphasised in which context the succession is performed).

Reply:
I see your point. I did this to be closer to the original code, but maybe it doesn't help too much (besides explaining what we do by the means of the function name, instead of using a comma, which seems to be our coding style). Regarding your proposed `succeedOwnSharedThirdPartyRelationshipAttribute` I don't see the need to define a function that does exactly the same than an already existing function does (and as I explained above I also don't see the need to distinguish between own shared RelationshipAttributes and own shared ThirdPartyRelationshipAttributes).

Reply:
Yes, that's right, because of the extra definitions of the functions you can determine what they do by their names. My suggestion would probably only shift the problem with the redundant code sections to the AttributesController.ts, but not solve it if we still want to have descriptive names. Maybe you have another idea for this.